win32ss/user/ntuser/winpos.c

```diff
@@ -2100,8 +2100,15 @@ co_WinPosSetWindowPos(
 
                 if ( (Window->style & WS_CHILD) && (Parent) && !(Parent->style & WS_CLIPCHILDREN))
                 {
-                   IntInvalidateWindows( Parent, DirtyRgn, RDW_ERASE | RDW_INVALIDATE);
-                   co_IntPaintWindows(Parent, RDW_NOCHILDREN, FALSE);
+                   /* HACK: For some reason, Rebar windows matches this condition after being altered
+                    * by ShowWindow() or UpdateWindow(), and as Rebar windows usually have children
+                    * Toolbars, these are not re-painted due to a missing RDW_ALLCHILDREN flag.
+                    * This might not have sense here, but fixes CORE-12342, CORE-15349, CORE-15917 and
+                    * very likely many others.
+                    */
+                   IntInvalidateWindows( Parent, DirtyRgn, RDW_ERASE | RDW_INVALIDATE | RDW_ALLCHILDREN);
```

Comment:
Please remove the space character before Parent, according to our coding-guidelines.

Then Please intentionally touch the 2nd IntInvalidateWindows() a few lines below in the else-branch the same way, so that it becomes visible in this review and will be visible also in a future git blame. To make it clearly visible that both code-paths will add RDW_ALLCHILDREN in the future!

And please readd in the comment that the change in Line2019 (adding RDW_ALLCHILDREN) does reduce the amount of executed tests of user32:msg_messages and user32:msg_winpos
like we had it clearly visible in
https://jira.reactos.org/secure/attachment/61682/61682_rebar_fixup4.patch

The future developers might be thankful for that information, because at some point in the future we will definitely have to touch that again, when something more correct was found.